### PR TITLE
[User accounts] Add project option to user accounts

### PR DESCRIPTION
**Pull request by ridz1208**
## Brief summary of changes
Add the functionality to affiliate users to specific projects.

At this stage affiliation with a project does not affect any of the data access, that will be added later on.

Changes in this PR were made to very closely resemble the CenterID/User interaction for consistency

#### Testing instructions (if applicable)

1. Try creating new users with projects
2. try adding/removing projects from existing users

This PR depends on the changes in #4919 for the changes in the Project table SQL. Here is the diff https://github.com/ridz1208/Loris/compare/2019_07_01_project_to_registration_project...ridz1208:2019-08_05_add_user_project

### CODE REVIEW

File: test/unittests/CandidateTest.php [hunk 1]
```diff
@@ -298,8 +298,8 @@ public function testGetProjectTitle()
         $this->_candidate->select($this->_candidateInfo['CandID']);
 
         $this->_dbMock->expects($this->any())
-            ->method('pselect')
-            ->willReturn($this->_listOfProjects);
+            ->method('pselectColWithIndexKey')
```
Comment (ridz1208) [inline test/unittests/CandidateTest.php]: @AlexandraLivadas @driusan had to chnage this one for it to pass travis

**Comment by driusan** [comment]
@ridz1208 Why is this marked as blocked?

**Comment by ridz1208** [comment]
Depends on 4919. See description

**Comment by cmadjar** [comment]
@Jkat Can you please review this PR before Thursday? Thank you!!

File: modules/user_accounts/php/edit_user.class.inc [hunk 2]
```diff
@@ -1116,6 +1159,15 @@ class Edit_User extends \NDB_Form
                 "site/affiliation";
         }
 
+        //======================================
+        //           Validate Project
+        //======================================
+        // Ensure that at least one site is selected
+        if (empty($values['ProjectIDs'])) {
+            $errors['projectss_group'] = "You must select at least one " .
```
Comment (Jkat) [inline modules/user_accounts/php/edit_user.class.inc]: `projects_group`

**Comment by Jkat** [comment]
Code changes look fine, my sandbox is still blocked because of database administration issues, unable to test frontend at the moment.

**Comment by Jkat** [comment]
I just tested it and it seems to work fine. My sandbox is kinda in a screwed up state though. But yeah should be just those 2 small fixes in my review.

File: modules/user_accounts/php/edit_user.class.inc [hunk 3]
```diff
@@ -705,6 +731,7 @@ class Edit_User extends \NDB_Form
         $siteOptions = array();
         $site        = array();
 
+        $siteOptions = array();
```
[the remark this answers is not in the document]
Comment (ridz1208) [reply modules/user_accounts/php/edit_user.class.inc]: i dont know

**Comment by ridz1208** [comment]
@Jkat fixed !!

**Comment by driusan** [comment]
This is going to fail Travis when it finishes

```
FILE: /home/travis/build/aces/Loris/php/libraries/User.class.inc

----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 307 | ERROR | [x] Whitespace found at end of line
 419 | ERROR | [x] Whitespace found at end of line
----------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

**Comment by christinerogers** [comment]
This PR was based on 
https://github.com/aces/Loris/pull/4255 
with work by Zain and especially Jacob Penny from the CAP project, and initially committed for Loris by Justin.

**Pull request merged**